packages/experimental/Avatar/src/JSAvatarTokens.ts

```diff
@@ -132,161 +132,132 @@ export const defaultJSAvatarTokens: TokenSettings<JSAvatarTokens, Theme> = (t: T
     },
     navy: {
-      backgroundColor: globalTokens.color.navy.tint40,
-      color: globalTokens.color.navy.shade30,
-      borderColor: globalTokens.color.navy.primary,
+      ...getColorProps('navy', t),
     },
     lavender: {
-      backgroundColor: globalTokens.color.lavender.tint40,
-      color: globalTokens.color.lavender.shade30,
-      borderColor: globalTokens.color.lavender.primary,
+      ...getColorProps('lavender', t),
     },
     purple: {
-      backgroundColor: globalTokens.color.purple.tint40,
-      color: globalTokens.color.purple.shade30,
-      borderColor: globalTokens.color.purple.primary,
+      ...getColorProps('purple', t),
     },
     grape: {
-      backgroundColor: globalTokens.color.grape.tint40,
-      color: globalTokens.color.grape.shade30,
-      borderColor: globalTokens.color.grape.primary,
+      ...getColorProps('grape', t),
     },
     lilac: {
-      backgroundColor: globalTokens.color.lilac.tint40,
-      color: globalTokens.color.lilac.shade30,
-      borderColor: globalTokens.color.lilac.primary,
+      ...getColorProps('lilac', t),
     },
     pink: {
-      backgroundColor: globalTokens.color.pink.tint40,
-      color: globalTokens.color.pink.shade30,
-      borderColor: globalTokens.color.pink.primary,
+      ...getColorProps('pink', t),
     },
     magenta: {
-      backgroundColor: globalTokens.color.magenta.tint40,
-      color: globalTokens.color.magenta.shade30,
-      borderColor: globalTokens.color.magenta.primary,
+      ...getColorProps('magenta', t),
     },
     plum: {
-      backgroundColor: globalTokens.color.plum.tint40,
-      color: globalTokens.color.plum.shade30,
-      borderColor: globalTokens.color.plum.primary,
+      ...getColorProps('plum', t),
     },
     beige: {
-      backgroundColor: globalTokens.color.beige.tint40,
-      color: globalTokens.color.beige.shade30,
-      borderColor: globalTokens.color.beige.primary,
+      ...getColorProps('beige', t),
     },
     mink: {
-      backgroundColor: globalTokens.color.mink.tint40,
-      color: globalTokens.color.mink.shade30,
-      borderColor: globalTokens.color.mink.primary,
+      ...getColorProps('mink', t),
     },
     platinum: {
-      backgroundColor: globalTokens.color.platinum.tint40,
-      color: globalTokens.color.platinum.shade30,
-      borderColor: globalTokens.color.platinum.primary,
+      ...getColorProps('platinum', t),
     },
     anchor: {
-      backgroundColor: globalTokens.color.anchor.tint40,
-      color: globalTokens.color.anchor.shade30,
-      borderColor: globalTokens.color.anchor.primary,
+      ...getColorProps('anchor', t),
     },
   } as JSAvatarTokens);
+
+/**
+ * A function which returns object of props depending on color and theme.
+ * @param color
+ * @param theme
+ * @returns object of props - backgroundColor, color and borderColor
+ */
+function getColorProps(color: string, theme: Theme) {
+  const themeAppearance = theme.host.appearance;
```

Comment:
This works for UWP but won't work for win32, since we use different names for Themes. MacOS will need a different implementation as well (@chiuam and @saadnajmi)

You can look at https://github.com/microsoft/fluentui-react-native/blob/master/packages/theming/win32-theme/src/getOfficeAliasTokens.ts and its implementor as an example of how to look for the current win32 theme.